### PR TITLE
Ensure digest; Garbage collect reqs, version check from FileServer

### PR DESCRIPTION
**Pull request by ArnobKumarSaha**
### general
- `DownloadedAt`, `NextUpdate` fields remove from trivyVersion
- `ImageDetails` struct moved to `trivy` pkg to use that in both `ImageScanRequest` & `BackendResponse`
- Always ensure image digest in ScanRequest's status, & ScanReport's spec.  For public, digest came from crane.Digest. For private, digest came from `containerStatus` of the private image pod.
- All the `RequeueAfter: TrivtDBRefreshPeriod` things removed. 

### fileServer-related
- Pass `fileServerDir`, `GarbageCollectionPeriod` in the Reconciler.
- MaxUploadsize for fileServer is increased to 1GB
- Read TrivyDBVersion from fs to check if reconciliation is needed.

### CODE REVIEW

File: pkg/controllers/private.go [hunk 1]
```diff
@@ -46,6 +54,11 @@ const (
 	containerUploader = "uploader"
 )
 
+var podSelectors = map[string]string{
```
Comment (tamalsaha) [inline pkg/controllers/private.go]: why?

File: pkg/controllers/reconciler.go [hunk 2]
```diff
@@ -36,60 +35,70 @@ import (
...
 	}
 	rr := RequestReconciler{
 		Reconciler: r,
+		ctx:        ctx,
 		req:        &scanreq,
 	}
 
-	if !rr.isReconciliationNeeded() {
-		return ctrl.Result{RequeueAfter: backend.TrivyRefreshPeriod}, nil
+	if time.Since(scanreq.CreationTimestamp.Time) > r.garbageCollectionPeriod {
```
Comment (tamalsaha) [inline pkg/controllers/reconciler.go]: Only requests that are complete (Success or Failure)

**Comment by tamalsaha** [comment]
- [x] job pod labels
- [x] show reports with out-dated report
- [x] Request.status.image field
- [x] only generate request if db is fresh (ui-server)
- [ ] Increase workload controller resync period to 1 hr
- [x] remove workload credential cache from ui-server

**Review by ArnobKumarSaha — COMMENTED** [review]
If a scanReq phase is `Current`, new fresh-scanning will never be happened, as we are returning from [here](https://github.com/kubeops/scanner/blob/4dd1a6b24e90b39a20d298cbc85434c2ce8d3bbe/pkg/controllers/reconciler.go#L94).  In this case, that scanReq is just waiting to be garbage collected. Is this the flow we are going to achieve ?

**Comment by tamalsaha** [comment]
> If a scanReq phase is `Current`, new fresh-scanning will never be happened, as we are returning from [here](https://github.com/kubeops/scanner/blob/4dd1a6b24e90b39a20d298cbc85434c2ce8d3bbe/pkg/controllers/reconciler.go#L94). In this case, that scanReq is just waiting to be garbage collected. Is this the flow we are going to achieve ?

Yes.

ScanRequests are one and done.

It is the ui-server's workflow controller's job to create new request if there is new db available.

**Pull request merged**